drivers/include/lorabee.h

```diff
@@ -0,0 +1,149 @@
+    LORABEE_ERR_NOT_JOINED,            /**< Network is not joined */
+    LORABEE_ERR_NO_FREE_CH,            /**< All channels are busy */
+    LORABEE_ERR_SILENT,                /**< Device is in Silent Immediately state */
+    LORABEE_ERR_FR_CNT_REJOIN_NEEDED,  /**< Frame counter rolled over */
+    LORABEE_ERR_BUSY,                  /**< MAC is not in Idle state */
+    LORABEE_ERR_INVALID_DATA_LENGTH,   /**< Wrong payload given */
+    LORABEE_ERR_KEYS_NOT_INIT,         /**< Keys not configured ("mac join" command) */
+};
+
+/**
+ * @brief   Configuration parameters for LoRaBee devices
+ */
+typedef struct {
+    uart_t uart;                       /**< UART interfaced the device is connected to */
+    uint32_t baudrate;                 /**< baudrate to use */
+    gpio_t pin_reset;                  /**< GPIO pin that is connected to the STATUS pin
+                                            set to GPIO_UNDEF if not used */
+} lorabee_params_t;
+
+/**
+ * @brief   LoRaBee device descriptor
+ */
+typedef struct {
+    /* device driver specific fields */
+    lorabee_params_t p;                        /**< configuration parameters */
+
+    /* values for the UART TX state machine  */
+    mutex_t cmd_lock;                          /**< mutex to allow only one
+                                                *   command at a time */
+
+    /* buffer and synchronization for command responses */
+    mutex_t resp_lock;                         /**< mutex for waiting for command
+                                                *   response */
+    uint8_t resp_buf[LORABEE_MAX_MSG_LENGTH]; /**< command response data buffer */
+    uint16_t resp_size;                       /**< counter for received char in response */
+    uint8_t resp_done;                        /**< check if response has completed */
+} lorabee_t;
+
+/**
+ * @brief   Prepare the given LoRaBee device
+ *
+ * @param[out] dev          LoRaBee device to initialize
+ * @param[in]  params       parameters for device initialization
+ *
+ * @return                  0 on success
+ * @return                  -ENODEV on invalid device descriptor
+ * @return                  -ENXIO on invalid UART or GPIO pins
+ */
+void lorabee_setup(lorabee_t *dev, const lorabee_params_t *params);
+
+/**
+ * @brief   Initializes the LoRaBee device
+ *
+ * @param[in] dev           LoRaBee device descriptor
+ *
+ * @return                  0 on success
+ * @return                  -ENXIO if UART initialization failed
+ */
+int lorabee_init(lorabee_t *dev);
+
+/**
+ * @brief   Writes a command to the LoRaBee device
+ *
+ * The module will immediately reply with a meaningful message if the command
+ * is valid or not.
+ *
+ * @param[in] dev           LoRaBee device descriptor
+ * @param[in] cmd           Command to send as string
+ *
+ * @return                  LORABEE_OK on success
+ * @return                  -LORABEE_ERR_INVALID_PARAM if command is invalid
```

Comment:
Why not avoid the minus here (and in _parse_response) and make the enum values explicit negative?

Reply:
because I don' t have to explicitly set a value to all enum values. I don't know what's best I admit.

Reply:
But then I have to ask, why do you need negative numbers at all? Why not just compare for != OK when you want to know if there is an error (instead of looking for negative numbers)?